src/spring-cloud/azext_spring_cloud/_validators.py

```diff
@@ -420,3 +418,21 @@ def _validate_resource_group_name(name, message_name):
     matchObj = match(r'^[-\w\._\(\)]+$', name)
     if matchObj is None:
         raise CLIError('--{0} must conform to the following pattern: \'^[-\\w\\._\\(\\)]+$\'.'.format(message_name))
+
+
+def _validate_route_table(namespace, vnet_obj):
+    app_route_table_id = ""
+    runtime_route_table_id = ""
+    for subnet in vnet_obj.subnets:
+        if subnet.id.lower() == namespace.app_subnet.lower() and subnet.route_table:
+            app_route_table_id = subnet.route_table.id
+        if subnet.id.lower() == namespace.service_runtime_subnet.lower() and subnet.route_table:
+            runtime_route_table_id = subnet.route_table.id
+
+    if app_route_table_id != "" and runtime_route_table_id != "":
+        if app_route_table_id == runtime_route_table_id:
+            raise CLIError('--service-runtime-subnet and --app-subnet should associate with different route tables.')
+    if app_route_table_id != "" and runtime_route_table_id == "" \
+            or app_route_table_id == "" and runtime_route_table_id != "":
+        raise CLIError(
+            '--service-runtime-subnet and --app-subnet should both associate with different route tables or neither.')

```

Comment:
confusing...

Reply:
The two subnets should both bind with different route tables or neither.